README.md

```diff
@@ -1,2 +1,56 @@
 # envoy-go-basic-auth
-Basic Auth based on Envoy Golang http filter
+
+This is a simple basic auth filter for envoy written in go. Only requests that pass the configuration authentication will be proxied to the upstream service.
+
+## Status
+
+This is under active development and is not ready for production use.
+
+## Usage
+
+The client set credentials in `Authorization` header in the following format:
+
+```Plaintext
+credentials := Basic base64(username:password)
+```
+
+An example of the `Authorization` header is as follows (`Zm9vOmJhcg==`, which is the base64-encoded value of `foo:bar`):
+
+```Plaintext
+Authorization: Basic Zm9vOmJhcg==
+```
+
+Configure your envoy.yaml to set pairs of username and password.
```

Comment:
The `make run` will use the `envoy. yaml` in this project. Maybe we can provide a link here so people can know which configuration will be used.